apps/snitch_core/test/data/model/package_item_test.exs

```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
-               PackageItem.create(%{
-                 @params
-                 | line_item_id: line_item.id,
-                   product_id: line_item.product_id,
-                   package_id: make_package(context).id
-               })
+    test "with invalid attributes" do
+      {:error, cs} =
+        PackageItem.create(%{
+          @params
+          | line_item_id: 2,
+            product_id: 3,
+            package_id: -2
+        })
 
-      assert package_item.backordered?
+      assert %{package_id: ["does not exist"]} == errors_on(cs)
     end
   end
 
   describe "update/2" do
     @tag variant_count: 1,
-         shipping_category_count: 1,
-         shipping_method_count: 1,
-         state_zone_count: 1
+           shipping_category_count: 1,
+           shipping_method_count: 1,
+           state_zone_count: 1
+    test "successful with valid params", context do
+      %{package_item: package_item} = new_package(context)
+      params = %{quantity: 95}
+      {:ok, updated_package_item} = PackageItem.update(params, package_item)
+      assert updated_package_item.id == package_item.id
+      assert updated_package_item.quantity != package_item.quantity
+      refute updated_package_item.backordered?
+    end
 
-    test "with valid params", context do
-      %{line_items: [line_item]} = context
+    test "with invalid params", context do
+      %{package_item: package_item} = new_package(context)
+      money = Money.new(-1, :USD)
+      updates = %{tax: money }
+      {:error, updated} = PackageItem.update(updates, package_item)
+      assert %{tax: ["must be equal or greater than 0"]} == errors_on(updated)
+    end
+  end
 
-      assert {:ok, package_item} =
-               PackageItem.create(%{
-                 @params
-                 | line_item_id: line_item.id,
-                   product_id: line_item.product_id,
-                   package_id: make_package(context).id
-               })
+  describe "delete/1" do
+    test "a package_item", context do
+      %{package_item: package_item} = new_package(context)
+      {:ok, _} = PackageItem.delete(package_item)
+      assert {:error, :package_item_not_found} == PackageItem.get(package_item.id)
+    end
+  end
 
-      assert package_item.backordered?
+  describe "get/1" do
+    test "with non-negative integer", context do
+      %{package_item: package_item} = new_package(context)
+      {:ok, new_package_item} = PackageItem.get(package_item.id)
+      assert new_package_item.id == package_item.id
+    end
 
-      assert {:ok, package_item} =
-               PackageItem.update(
-                 %{
-                   state: "destroyed",
-                   quantity: 5,
-                   delta: 0
-                 },
-                 package_item
-               )
+    test "with a map", context do
+      %{package_item: package_item} = new_package(context)
+      map = %{quantity: package_item.quantity, delta: package_item.delta}
```

Comment:
Use something which is unique for `package item` for e.g. package number. The keys that you have used are not unique for a package item. You have used quantity and delta do you think these are unique for `package items`